packages/providers/test/retry.test.ts

```diff
@@ -0,0 +1,138 @@
+    // this might be something related to the event loop and how it schedules setTimeouts
+    // expectedWaitTime minus 5ms seems like reasonable to allow
+    expect(waitTime).toBeGreaterThanOrEqual(expectedWaitTime - 5);
+    expect(waitTime).toBeLessThanOrEqual(expectedWaitTime + 10);
+  }
+
+  test('fixed backoff', async () => {
+    const retryOptions = { maxAttempts, baseDuration: duration, backoff: 'fixed' as const };
+
+    const provider = await Provider.create(FUEL_NETWORK_URL, { retryOptions });
+    const callTimes: number[] = [];
+
+    mockFetch(maxAttempts, callTimes);
+
+    const expectedChainInfo = await provider.operations.getChain();
+
+    const chainInfo = await provider.operations.getChain();
+
+    expect(chainInfo).toEqual(expectedChainInfo);
+    expect(callTimes.length - 1).toBe(maxAttempts); // callTimes.length - 1 is for the initial call that's not a retry so we ignore it
+
+    callTimes.forEach((callTime, index) => assertBackoff(callTime, index, callTimes, duration));
+  });
+
+  test('linear backoff', async () => {
+    const retryOptions = { maxAttempts, baseDuration: duration, backoff: 'linear' as const };
+
+    const provider = await Provider.create(FUEL_NETWORK_URL, { retryOptions });
+    const callTimes: number[] = [];
+
+    mockFetch(maxAttempts, callTimes);
+
+    const expectedChainInfo = await provider.operations.getChain();
+
+    const chainInfo = await provider.operations.getChain();
+
+    expect(chainInfo).toEqual(expectedChainInfo);
+    expect(callTimes.length - 1).toBe(maxAttempts); // callTimes.length - 1 is for the initial call that's not a retry so we ignore it
+
+    callTimes.forEach((callTime, index) =>
+      assertBackoff(callTime, index, callTimes, duration * index)
+    );
+  });
+
+  test('exponential backoff', async () => {
+    const retryOptions = { maxAttempts, baseDuration: duration, backoff: 'exponential' as const };
+
+    const provider = await Provider.create(FUEL_NETWORK_URL, { retryOptions });
+    const callTimes: number[] = [];
+
+    mockFetch(maxAttempts, callTimes);
+
+    const expectedChainInfo = await provider.operations.getChain();
+
+    const chainInfo = await provider.operations.getChain();
+
+    expect(chainInfo).toEqual(expectedChainInfo);
+    expect(callTimes.length - 1).toBe(maxAttempts); // callTimes.length - 1 is for the initial call that's not a retry so we ignore it
+
+    callTimes.forEach((callTime, index) =>
+      assertBackoff(callTime, index, callTimes, duration * (2 ^ (index - 1)))
+    );
+  });
+
+  test('throws if last attempt fails', async () => {
+    const retryOptions = { maxAttempts, baseDuration: duration, backoff: 'fixed' as const };
+
+    const provider = await Provider.create(FUEL_NETWORK_URL, { retryOptions });
+
+    const fetchSpy = jest
+      .spyOn(global, 'fetch')
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore TS is throwing error when test is run, but not in IDE
```

Comment:
You must tell `providers/tsconfig.json` to consider your `test` directory.

```json
{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist"
  },
  "include": ["src", "test"]
}

```

Reply:
This was a copy-paste linting ignore that I said to myself "I'll fix it later"... but the later never came. :D